src/common/reducers/entities/snap.js

```diff
@@ -1,11 +1,50 @@
 import * as RegisterNameActionTypes from '../../actions/register-name';
 
 export default function snap(state={}, action) {
+  const { payload } = action;
+
+  const registerNameInitialStatus = {
+    isFetching: false,
+    success: false,
+    error: null
+  };
+
   switch(action.type) {
+    case RegisterNameActionTypes.REGISTER_NAME:
+      return {
+        ...state,
+        registerNameStatus: {
+          ...registerNameInitialStatus,
+          isFetching: true
```

Comment:
I wasn't sure if this should be flat in snap object or better nested in custom registerNameStatus, but success and error seem quite custom here (error representing error of register name action or validation of register name), so I think it's safer to keep it in custom namespace.